lib/travis/api/v3/service.rb

```diff
@@ -89,21 +89,23 @@ def result_type
       self.class.result_type
     end
 
-    def result(*args)
-      Result.new(access_control, *args)
+    def result(resource, **meta_data)
+      return not_found unless resource
+      rt = meta_data[:result_type] || result_type
+      status = meta_data[:status] || 200
+      Result.new(access_control: access_control, type: rt, resource: resource, status: status, **meta_data)
     end
 
-    def head(*args)
-      Result::Head.new(access_control, *args)
+    def head
+      Result::Head.new(access_control: access_control, type: result_type, resource: nil, status: 204)
```

Comment:
I think `head` and `deleted` should be different – `head` should allow for other statuses besides 204.